crates/matrix-sdk-sqlite/src/utils.rs

```diff
@@ -280,6 +281,19 @@ impl SqliteKeyValueStoreConnectionExt for rusqlite::Connection {
 /// ```
 #[async_trait]
 pub(crate) trait SqliteKeyValueStoreAsyncConnExt: SqliteAsyncConnExt {
+    /// Whether the `kv` table exists in this database.
+    async fn kv_table_exists(&self) -> rusqlite::Result<bool> {
+        let count = self
+            .query_row(
+                "SELECT count(*) FROM sqlite_master WHERE type = 'table' AND name = 'kv'",
```

Comment:
I believe `SELECT 1` would be faster than `SELECT count(*)` here. What do you think?

Reply:
In your case that means that we should check whether it returns a row or not, right?

I have been wondering if we should do this:

```sql
SELECT EXISTS (SELECT 1 FROM sqlite_master WHERE type = 'table' AND name = 'kv')
```

Which will basically return a boolean.

Reply:
Yeah, that's another approach and I like it too.

Reply:
Done